src/bun.js/bindings/Script.cpp

```diff
@@ -0,0 +1,181 @@
+        return ptr;
+    }
+
+    DECLARE_INFO;
+    template<typename CellType, SubspaceAccess>
+    static GCClient::IsoSubspace* subspaceFor(VM& vm)
+    {
+        return &vm.plainObjectSpace();
+    }
+    static Structure* createStructure(VM& vm, JSGlobalObject* globalObject, JSValue prototype)
+    {
+        return Structure::create(vm, globalObject, prototype, TypeInfo(ObjectType, StructureFlags), info());
+    }
+
+private:
+    ScriptPrototype(VM& vm, JSGlobalObject* globalObject, Structure* structure)
+        : Base(vm, structure)
+    {
+    }
+
+    void finishCreation(VM&, JSGlobalObject*);
+};
+STATIC_ASSERT_ISO_SUBSPACE_SHARABLE(ScriptPrototype, ScriptPrototype::Base);
+
+static const struct HashTableValue scriptPrototypeTableValues[] = {
+   { "cachedDataRejected"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::CustomAccessor), NoIntrinsic, { HashTableValue::GetterSetterType, scriptGetCachedDataRejected, nullptr } },
+   { "createCachedData"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::Function), NoIntrinsic, { HashTableValue::NativeFunctionType, scriptCreateCachedData, 0 } },
+   { "runInContext"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::Function), NoIntrinsic, { HashTableValue::NativeFunctionType, scriptRunInContext, 0 } },
+   { "sourceMapURL"_s, static_cast<unsigned>(PropertyAttribute::ReadOnly|PropertyAttribute::CustomAccessor), NoIntrinsic, { HashTableValue::GetterSetterType, scriptGetSourceMapURL, nullptr } },
+};
+
+const ClassInfo ScriptPrototype::s_info = { "Script"_s, &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(ScriptPrototype) };
+const ClassInfo Script::s_info = { "Script"_s, &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(Script) };
+const ClassInfo ScriptConstructor::s_info = { "Script"_s, &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(ScriptConstructor) };
+
+ScriptConstructor::ScriptConstructor(VM& vm, Structure* structure)
+    : ScriptConstructor::Base(vm, structure, scriptConstructorCall, scriptConstructorConstruct)
+{
+}
+
+ScriptConstructor* ScriptConstructor::create(VM& vm, JSGlobalObject* globalObject, Structure* structure, JSObject* prototype)
+{
+    ScriptConstructor* ptr = new (NotNull, allocateCell<ScriptConstructor>(vm)) ScriptConstructor(vm, structure);
+    ptr->finishCreation(vm, globalObject, prototype);
+    return ptr;
+}
+
+void ScriptConstructor::finishCreation(VM& vm, JSGlobalObject* globalObject, JSObject* prototype)
+{
+    Base::finishCreation(vm);
+    ASSERT(inherits(info()));
+    initializeProperties(vm, globalObject, prototype);
+}
+
+void ScriptConstructor::initializeProperties(VM& vm, JSGlobalObject* globalObject, JSObject* prototype)
+{
+    putDirect(vm, vm.propertyNames->length, jsNumber(0), PropertyAttribute::ReadOnly | PropertyAttribute::DontEnum);
+    JSString* nameString = jsNontrivialString(vm, "Script"_s);
+    m_originalName.set(vm, this, nameString);
```

Comment:
This might be optional but isn't harmful

This pattern of `initializeProperties` largely exists for the DOMStructureMap stuff

We won't need this here

but it's nbd

Reply:
I just copypasted this from somewhere and forgot to go back to make sure I needed it haha. Happy to remove it